Fixed the bug in the fit method of the EpochSelectChannel class

allwaysFindFood opened the pull request:
Fixed the bug in the fit method of the EpochSelectChannel class, where it failed to identify the top N channels.
For detailed information, please visit  https://github.com/NeuroTechX/moabb/issues/683#issuecomment-2642593588

bruAristimunha commented:
Thank you @allwaysFindFood!

bruAristimunha commented:
Can you please update the whats_new file? I think @gcattan is the best person to review this pr.

gcattan commented on examples/plot_Hinss2021_classification.py:
```diff
@@ -65,22 +65,12 @@ def fit(self, X, _y=None):
-        indices = []
-        for v in all_max:
-            indices.extend(np.argwhere(m == v).flatten())
+        indices = np.unravel_index(
+            np.argpartition(m, -self.n_chan, axis=None)[-self.n_chan :], m.shape
+        )
```
Thank you for your PR @allwaysFindFood :)

I am trying this change in the console, with a small example.

![image](https://github.com/user-attachments/assets/385565a6-114e-4509-8650-dcc0f633744e)

Shouldn't be returning [1, 1] and [1, 0] instead?

allwaysFindFood replied:
This return type seems to have no impact on the code, so I didn't change “indices” and   "self._chs_idx" to the list type.

gcattan replied:
I mean [0, 1] is 2. It should be returning [1, 0] that is 3. 
Probably needs to reverse sort it first
`np.argpartition(m, -self.n_chan, axis=None)[-self.n_chan :][::-1]` or something like this.

allwaysFindFood replied:
Apologies for the  misunderstanding. The first array returned by np.unravel_index represents the first dimension (e.g., rows), and the second array represents the second dimension (e.g., columns), so the returned information is correct，as shown below.
<img width="257" alt="1739154386756" src="https://github.com/user-attachments/assets/6972d890-0d5a-4c65-8dc2-ee8c2b88916f" />





Fortunately, since the m is a symmetric matrix, the order of rows and columns  does not matter, and there is no need to modify the code.

gcattan commented:
Thanks for the thorough investigation @allwaysFindFood .
Please, update the what new file: https://github.com/allwaysFindFood/moabb/edit/fix-EpochSelectChannel-bug/docs/source/whats_new.rst, and we are good to merge:

![image](https://github.com/user-attachments/assets/71dacb3f-0c52-4377-bdc5-57380a29e4e8)

You can add your Github handle at the end of the page: 

![image](https://github.com/user-attachments/assets/c05cf3bd-117b-4823-b5ec-19fe3ddcb72a)

Please, also consider making a contribution to [pyRiemann-qiskit](https://github.com/pyRiemann/pyRiemann-qiskit/blob/17987c8984b3e6d97104b4a38186855bb9da334b/pyriemann_qiskit/utils/filtering.py#L118), as the current code was duplicated there.

I can do it, but you may want to take credit for your fix there too.

allwaysFindFood commented:
@gcattan I have updated the whats_new.rst file and added my GitHub handle at the end of the page.  I used the abbreviation of my GitHub username . Let me know if you prefer a different format,  Regarding the contribution to pyRiemann-qiskit, I will address this in the near future and submit a separate PR. 
![image](https://github.com/user-attachments/assets/6f36b37a-2511-4126-83a9-91ee059017c0)

bruAristimunha commented:
Hey @allwaysFindFood, we usually use the real nome here, but if you want to keep anonymous, no problem, your contribution is welcome 🤗

bruAristimunha commented:
Thank you, Sijie Zhou (@allwaysFindFood); your contribution is always welcome, built on open-source, we can go further as a community! please don't forget to cite moabb in your publication, it is important for the maintainers, who do the work voluntarily and without funding :)

Thank you @gcattan for the review ;)

gcattan commented:
Yes, thank you for your contribution @allwaysFindFood :)
